### ISSUE TITLE
BUG: IRS practitioner sees incorrect service preference when address is sealed

### ISSUE THREAD

**Issue by klohman** [ustaxcourt/ef-cms#7745]
**Describe the Bug**
When a petitioner address is sealed, the petitioner's service indicator is displaying as "Paper" for IRS attorneys (associated and not associated) and the petitioner when the petitioner actually has electronic service. Internal users still see "Electronic" as service preference. 

**Business Impact/Reason for Severity**

**In which environment did you see this bug?**
migration 

**Who were you logged in as?**
irsPractitioner4@example.com
docketclerk1@example.com

**What were you doing when you discovered this bug? (Using the application, demoing, smoke tests, testing other functionality, etc.)**
PO testing 7807 

**To Reproduce**
1. Login as docket clerk 
2. Go to case with petitioner with electronic service 
3. Seal address 
4. Login as irs practitioner 
5. Go to case ( from step 2) 
6. View petitioner contact info 

**Expected Behavior**
Service preference should not change when address is sealed. IRS practitioner should see accurate service preference for petitioner's with electronic service and sealed address.

**Actual Behavior**
the petitioner's service indicator is displaying as "Paper" for IRS attorneys (associated and not associated) when the petitioner actually has electronic service. Internal users still see "Electronic" as service preference. 

**Screenshots**
IRS practitioner views "Paper" service 
![Screen Shot 2021-03-03 at 11.04.43 AM.png](https://images.zenhubusercontent.com/5bd7271508843955cc24d3d0/6bb3b77b-6b63-4837-8177-0049621015cb)

Docket clerk sees electronic service
![Screen Shot 2021-03-03 at 11.05.10 AM.png](https://images.zenhubusercontent.com/5bd7271508843955cc24d3d0/2f03e0c5-3791-4284-a18e-af0af9fdac0f)

**Desktop (please complete the following information):**

**Smartphone (please complete the following information):**

**Cause of Bug, If Known**

**Process for Logging a Bug:**
* Complete the above information
* Add a severity tag (Critical, High Severity, Medium Severity or Low Severity). See below for priority definition. 

**Severity Definition:**
* Critical Defect
Blocks entire system's or module’s functionality
No workarounds available
Testing cannot proceed further without bug being fixed.

* High-severity Defect
Affects key functionality of an application
There's a workaround, but not obvious or easy
App behaves in a way that is strongly different from the one stated in the requirements

* Medium-severity Defect
A minor function does not behave in a way stated in the requirements.
Workaround is available and easy

* Low-severity Defect
Mostly related to an application’s UI
Doesn't need a workaround, because it doesn't impact functionality

## Definition of Done (Updated 2-23-21)
**Product Owner**
 - [x]  Acceptance criteria have been met and validated on the Flexion Prod env

**UX**
 - [ ] Business test scenarios to meet all acceptance criteria have been written
 - [x] Usability has been validated
 - [ ] Wiki has been updated (if applicable) 
 - [ ] Story has been tested on a mobile device (for external users only)
 - [ ] Add scenario to testing document, if applicable (https://docs.google.com/spreadsheets/d/1FUHKC_YrT-PosaWD5gRVmsDzI1HS_U-8CyMIb-qX9EA/edit?usp=sharing)

**Engineering**
 - [x] Automated test scripts have been written
 - [x] Field level and page level validation errors (front-end and server-side) integrated and functioning
 - [x] Verify that language for docket record for internal users and external users is identical
 - [x] New screens have been added to pa11y scripts
 - [x] All new functionality verified to work with keyboard and macOS voiceover https://www.apple.com/voiceover/info/guide/_1124.html 
 - [x] READMEs, other appropriate docs, JSDocs and swagger/APIs fully updated
 - [x] UI should be touch optimized and responsive for external only (functions on supported mobile devices and optimized for screen sizes as required)
 - [x] Module dependencies are up-to-date and are at the latest resolvable version (npm update)
 - [x] Errors in Sonarcloud are fixed https://sonarcloud.io/organizations/flexion-github/projects
 - [x] Lambdas include CloudWatch logging of users, inputs and outputs
 - [x] Interactors should validate entities before calling persistence methods
 - [x] Code refactored for clarity and to remove any known technical debt
 - [x] Rebuild entity documentation
 - [x] Acceptance criteria for the story has been met
 - [x] Deployed to the dev environment
 - [x] Deployed to the Court's migration environment

 **Review Steps**
 1. Finish all other DOD
 2. Deploy to the dev environment
 3. Engineers add `Needs UX Review` label
 4. UX Review on dev environment (if feedback, implement and go back to step 2)
 5. UX add `Needs Migration Deploy` label
 6. Deploy to the Court's migration environment
 7. Engineers go through test scenarios on Court's migration environment
 8. Engineers add `Needs PO Review` label and move to Review/QA column
 9. PO review (if feedback, implement and go back to step 2)

**Comment by klohman** [comment]
Similar bug - https://github.com/flexion/ef-cms/issues/7717

**Comment by matthopson** [comment]
@klohman Just a note for reference. The main issue was when a contactPrimary had a service preference set, it wasn't being conveyed to non-associated users, so the fallbacks would kick in (the logic that determines the service preference). So for a paper case (without representation) and a sealed address, it would default to paper, EVEN IF the contactPrimary.serviceIndicator was set to Electronic because we limit what fields are available to the user on a sealed address. The fix was to ensure the serviceIndicator field was added in the allowed fields that come through the case filter for sealed addresses. We wrote some integration tests that run through various scenarios where a service preference might be changed and viewed by various users to ensure expectations are always consistent.
